doc/languages-frameworks/php.section.md

```diff
@@ -153,3 +155,144 @@ php.override {
   };
 }
 ```
+
+### Building PHP projects
+
+With [Composer](https://getcomposer.org/), you can effectively build PHP
+projects by streamlining dependency management. As the de-facto standard
+dependency manager for PHP, Composer enables you to declare and manage the
+libraries your project relies on, ensuring a more organized and efficient
+development process.
+
+Composer is not a package manager in the same sense as `Yum` or `Apt` are. Yes,
+it deals with "packages" or libraries, but it manages them on a per-project
+basis, installing them in a directory (e.g. `vendor`) inside your project. By
+default, it does not install anything globally. This idea is not new and
+Composer is strongly inspired by node's `npm` and ruby's `bundler`.
+
+Currently, there is no other PHP tool that offers the same functionality as
+Composer.
+
+In a Composer project, dependencies are defined in a `composer.json` file,
+while their specific versions are locked in a `composer.lock` file. Some
+Composer-based projects opt to include this `composer.lock` file in their source
+code, while others choose not to.
+
+In Nix, there are multiple approaches to building a Composer-based project.
+
+One such method is the `php.buildComposerProject` function, which serves as a
+wrapper around `mkDerivation`.
+
+Using this function, you can build a PHP project that includes both a
+`composer.json` and `composer.lock` file. If the project specifies binaries
+using the `bin` attribute in `composer.json`, these binaries will be
+automatically linked and made accessible in the derivation.
+
+To use the wrapper effectively, simply add the `vendorHash` attribute, which
+enables the wrapper to handle the heavy lifting.
+
+Internally, the wrapper operates in three stages:
+
+1. It constructs a `vendor cache` derivation by downloading the dependencies
+   specified in `composer.json`. This process uses the function
+   `php.buildComposerDeps` which in turn uses the
+   `php.composerHooks.composerSetupHook` hook.
+2. The resulting `vendor cache` derivation is then used by the
+   `php.composerHooks.composerInstallHook` hook, which is responsible for
+   creating the final `vendor` directory from the temporary `vendor cache`
+   directory.
+3. Any binary specified in the `composer.json` are linked and made accessible in
+   the derivation.
+
+By default, the Composer commands adds the following command line flags:
+
+ - `--no-dev`: User configurable, skip installing packages listed in
+  `require-dev`
```

Comment:
```suggestion
 - `--no-dev` to skip installing packages listed in `require-dev`, pass `installComposerDevDependencies = true;` to override.
```

Reply:
Or alternately, focus on the arguments, we are not composer docs

```suggestion
 - `installComposerDevDependencies`: set to `true` to not pass `--no-dev` to `composer install`.
```

Reply:
fixed, for the 3 of them